src/Lykke.Job.TransactionHandler.Services/InMemoryDeduplicator.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using Lykke.Job.TransactionHandler.Core.Services;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Lykke.Job.TransactionHandler.Services
+{
+    public class InMemoryDeduplicator : IDeduplicator
+    {
+        private static readonly TimeSpan DefaultCacheInvalidationTimeout = TimeSpan.FromMinutes(10);
+        private readonly IMemoryCache _cache;
```

Comment:
Probably it will be better to use now azure storage as persistance for deduplication.
As on tx handler restart / update we can accept a duplicate.